olive/passes/pytorch/lora.py

```diff
@@ -114,43 +116,34 @@ def create_training_args(self) -> transformers.TrainingArguments:
         return transformers.TrainingArguments(**args, **extra_args)
 
 
-class QLoRA(Pass):
-    """Run QLoRA fine-tuning on a Hugging Face PyTorch model.
-
-    See https://arxiv.org/abs/2305.14314 for more details on the method.
-
-    This pass only supports PyTorchModel with hf_config.
-    """
+class LoRABase(Pass):
+    """Base class for LoRA and QLoRA fine-tuning passes."""
 
     @staticmethod
     def _default_config(accelerator_spec: AcceleratorSpec) -> Dict[str, PassConfigParam]:
         return {
-            # quantization parameters
-            "compute_dtype": PassConfigParam(
-                type_=str,
-                default_value="bfloat16",
-                description=(
-                    "The computation data type used by the quantized model. It is also the data type used for the LoRA"
-                    " weights. Should be one of `bfloat16`, `float16` or `float32`."
-                ),
+            "lora_r": PassConfigParam(type_=int, default_value=64, description="Lora attention dimension."),
+            "target_modules": PassConfigParam(type_=List[str], default_value=None, description="Target modules"),
+            "lora_alpha": PassConfigParam(
+                type_=float, default_value=16, description="The alpha parameter for Lora scaling."
             ),
-            "double_quant": PassConfigParam(
+            "lora_dropout": PassConfigParam(
+                type_=float, default_value=0.0, description="The dropout probability for Lora layers."
+            ),
+            "bias": PassConfigParam(type_=str, default_value="none", description="Bias type for Lora"),
+            "modules_to_save": PassConfigParam(
```

Comment:
should we expose `target_modules` and `modules_to_save` only in the actual `LoRA` pass? Since qlora ignores these parameters, it is confusing to expose them in the base config itself.

Reply:
Yeah I just noticed that qlora doesn't need this. I'll update this.